src/main/scala/io/keen/client/scala/Client.scala

```diff
@@ -381,6 +403,118 @@ trait Writer extends AccessLevel {
+        val executor = new ScheduledThreadPoolExecutor(1)
+        executor.scheduleAtFixedRate(new Runnable {
+          def run: Unit = {
+            sendQueuedEvents
+          }
+        }, 1, sendIntervalSeconds.toLong, TimeUnit.SECONDS)
+      }
+      // invalid interval
+      case k if k < MinSendInterval || k >= MaxSendInterval => {
+        throw new IllegalArgumentException(s"Send interval must be between $MinSendInterval and $MaxSendInterval")
+      }
+      // invalid interval value
+      case _ => {
+        throw new IllegalArgumentException(s"Invalid value specified for send interval. Value must be an integer. ")
+      }
+    }
+
+  }
+
+  /**
+   * Sends all queued events, removing events from the queue as events are successfully sent.
+   */
+  def sendQueuedEvents(): Unit = {
+
+    val handleMap: TrieMap[String, ListBuffer[Long]] = eventStore.getHandles(projectId)
+    val handles: ListBuffer[Long] = ListBuffer.empty[Long]
+    val events: ListBuffer[String] = ListBuffer.empty[String]
+
+    // iterate over all of the event handles in the queue, by collection
+    for((collection, eventHandles) <- handleMap) {
+      // get each event, and its handle, then add it to a buffer so we can group the events 
+      // into smaller batches
+      for(handle <- eventHandles) {
+        handles += handle
+        events += eventStore.get(handle)
+      }
+
+      // group handles separately so we can use them to remove events from the queue once they've
+      // been succesfully added
+      val handleGroup: List[ListBuffer[Long]] = handles.grouped(batchSize).toList
+
+      // group the events by batch size, then publish them
+      for((batch, index) <- events.grouped(batchSize).zipWithIndex) {
+        // publish this batch
+        var response = Await.result(
+          addEvents(s"""{"$collection": [${batch.mkString(",")}]}"""),
+          DurationInt(batchTimeout).seconds
+        )
+
+        // handle addEvents responses properly
+        response.statusCode match {
+          case 200 | 201 => {
+            // log success
+            Logger.kinfo(s"""${response.statusCode} ${response.body} | Sent ${batch.size} queued events""")
+
+            // remove all of the handles for this batch
+            for(handle <- handleGroup(index)) {
+              eventStore.remove(handle)
+            }
+
+            // log removal
+            Logger.kinfo(s"""Removed ${handleGroup(index).size} events from the queue""")
+          }
+          case _ => {
+            // log but DO NOT remove events from queue
+            Logger.kerror(s"""${response.statusCode} ${response.body} | Failed to send ${batch.size} queued events""") 
+          }
+        }
+      }
+    }
+
+  }
+
+  /**
+   * Asynchronously sends all queued events.
+   */
+  def sendQueuedEventsAsync(): Unit = {
+
+    val executor: Executor = Executors.newFixedThreadPool(Runtime.getRuntime.availableProcessors)
```

Comment:
You should also consider passing in an `ThreadFactory` that sets these threads to daemon threads so that they will not keep the program from exiting if the pools aren't shut down.